src/os/flatpak.zig

```diff
@@ -501,3 +501,61 @@ pub const FlatpakHostCommand = struct {
 
     fn noopCallback(_: ?*anyopaque, _: *xev.Loop, _: *Completion, _: WaitError!u8) void {}
 };
+
+/// A struct to help retrive information about the running Flatpak.
+///
+/// Requires GLib to be available and linked.
+pub const FlatpakInfo = struct {
+    const Self = @This();
+    const c = @cImport({
+        @cInclude("glib.h");
+    });
+
+    keyfile: *c.GKeyFile,
+    app_dir: ?[]u8,
+
+    pub fn init() !Self {
+        const keyfile = c.g_key_file_new();
+        errdefer c.g_key_file_unref(keyfile);
+
+        if (c.g_key_file_load_from_file(keyfile, "/.flatpak-info", 0, null) == 0) return error.FileNotFound;
+
+        return .{
+            .keyfile = keyfile.?,
+            .app_dir = null,
+        };
```

Comment:
There doesn't seem to be much point in keeping the keyfile around for the lifetime of the program. It would seem to me to be better to just read the key that we need, cache that, and then unref the keyfile.